embrace-android-core/src/main/kotlin/io/embrace/android/embracesdk/internal/injection/OpenTelemetryModuleImpl.kt

```diff
@@ -69,11 +70,19 @@ internal class OpenTelemetryModuleImpl(
         openTelemetrySdk.sdkTracer
     }
 
+    private var sensitiveKeysBehavior: SensitiveKeysBehavior? = null
+
+    override fun setupSensitiveKeysBehavior(sensitiveKeysBehavior: SensitiveKeysBehavior) {
+        this.sensitiveKeysBehavior = sensitiveKeysBehavior
+        embraceSpanFactory.setupSensitiveKeysBehavior(sensitiveKeysBehavior)
+    }
```

Comment:
I'm not convinced this is the right pattern & think it might be better to use a `() -> SensitiveKeysBehavior` parameter on the constructor of `OpenTelemetryModuleImpl` instead to get around any circular dependency. What was the reason `setupSensitiveKeysBehavior` was needed in the first place?

Reply:
Summing up our chat today here so Hanson can see it too: 

- setupSensitiveKeysBehavior is needed because OpenTelemetryModuleImpl is created before the LocalConfig.
- We can't pass a function that provides SensitiveKeysBehavior because the local config isn't initialized yet when the OpenTelemetryModuleImpl is constructed.
- This approach is good enough for now, but we should examine how the OpenTelemetryModule is initialized and see if it's worth deconstructing it into different parts.

Reply:
We really need to find a way for `LocalConfig` to be created before the initialization of the OTel SDK. It would not only help this, but also we need to properly populate the Resource attributes with the name and version of the app using the SDK, which we aren't do now.

With the `CodeGenConfig` work that @fractalwrench is playing with, we'll be able to do this, right?

Reply:
Yep - assuming we choose to go down that route.